File: bin/Debug/net5.0/v4.4.2/lgcy-on-lgcy/examples/t_api-captcha-mvc2-web.security.membership/csharp/CaptchaHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

using BotDetect;
using BotDetect.Web;
using BotDetect.Web.UI;
using BotDetect.Web.Mvc;

public class CaptchaHelper
{
    public static MvcCaptcha GetRegistrationCaptcha()
    {
        // all Captcha properties are set in this event handler
        WebFormsCaptcha.InitializedWebCaptcha +=
            new EventHandler<InitializedWebCaptchaEventArgs>(
                RegistrationCaptcha_InitializedWebCaptcha);

        // create the control instance
        MvcCaptcha registrationCaptcha = new MvcCaptcha("RegistrationCaptcha");
        registrationCaptcha.UserInputID = "CaptchaCode";

        return registrationCaptcha;
    }

    // event handler used for Captcha control property randomization
    public static void RegistrationCaptcha_InitializedWebCaptcha(object sender,
        InitializedWebCaptchaEventArgs e)
    {
        if (e.CaptchaId != "RegistrationCaptcha")
        {
            return;
        }

        Captcha registrationCaptcha = sender as Captcha;

        // fixed Captcha settings 
        registrationCaptcha.ImageSize = new System.Drawing.Size(200, 50);
        registrationCaptcha.CodeLength = 4;

        // randomized Captcha settings
        registrationCaptcha.ImageStyle = CaptchaRandomization.GetRandomImageStyle();
        registrationCaptcha.SoundStyle = CaptchaRandomization.GetRandomSoundStyle();
    }

    public static MvcCaptcha GetLoginCaptcha()
    {
        // all Captcha properties are set in this event handler
        WebFormsCaptcha.InitializedWebCaptcha +=
            new EventHandler<InitializedWebCaptchaEventArgs>(
                LoginCaptcha_InitializedWebCaptcha);

        // create the control instance
        MvcCaptcha loginCaptcha = new MvcCaptcha("LoginCaptcha");
        loginCaptcha.UserInputID = "CaptchaCode";

        return loginCaptcha;
    }

    // event handler used for Captcha control property randomization
    public static void LoginCaptcha_InitializedWebCaptcha(object sender,
        InitializedWebCaptchaEventArgs e)
    {
        if (e.CaptchaId != "LoginCaptcha")
        {
            return;
        }

        Captcha loginCaptcha = sender as Captcha;

        // fixed Captcha settings 
        loginCaptcha.ImageSize = new System.Drawing.Size(200, 50);
        loginCaptcha.CodeLength = 4;

        // randomized Captcha settings
        loginCaptcha.ImageStyle = CaptchaRandomization.GetRandomImageStyle();
        loginCaptcha.SoundStyle = CaptchaRandomization.GetRandomSoundStyle();
    }
}

File: bin/Debug/net5.0/v4.4.2/lgcy-on-lgcy/examples/t_api-captcha-mvc3-web.security.membership/vbnet/CaptchaHelper.vb
Imports BotDetect
Imports BotDetect.Web
Imports BotDetect.Web.UI
Imports BotDetect.Web.Mvc

Public Class CaptchaHelper

    Public Shared Function GetRegistrationCaptcha() As MvcCaptcha

        ' all Captcha properties are set in this event handler
        AddHandler WebFormsCaptcha.InitializedWebCaptcha, _
            AddressOf RegistrationCaptcha_InitializedWebCaptcha
        
        ' create the control instance
        Dim registrationCaptcha As BotDetect.Web.Mvc.MvcCaptcha = _
            New BotDetect.Web.Mvc.MvcCaptcha("RegistrationCaptcha")

        registrationCaptcha.UserInputID = "CaptchaCode"

        Return registrationCaptcha

    End Function

    ' event handler used for Captcha control property randomization
    Public Shared Sub RegistrationCaptcha_InitializedWebCaptcha( _
        ByVal sender As Object, _
        ByVal e As BotDetect.InitializedWebCaptchaEventArgs)

        If (e.CaptchaId <> "RegistrationCaptcha") Then
            Return
        End If

        Dim registrationCaptcha As Captcha = TryCast(sender, Captcha)

        ' fixed Captcha settings 
        registrationCaptcha.ImageSize = New System.Drawing.Size(200, 50)
        registrationCaptcha.CodeLength = 4

        ' randomized Captcha settings
        registrationCaptcha.ImageStyle = CaptchaRandomization.GetRandomImageStyle()
        registrationCaptcha.SoundStyle = CaptchaRandomization.GetRandomSoundStyle()
    End Sub

    Public Shared Function GetLoginCaptcha() As MvcCaptcha

        ' all Captcha properties are set in this event handler
        AddHandler WebFormsCaptcha.InitializedWebCaptcha, _
            AddressOf LoginCaptcha_InitializedWebCaptcha

        ' create the control instance
        Dim loginCaptcha As BotDetect.Web.Mvc.MvcCaptcha = _
            New BotDetect.Web.Mvc.MvcCaptcha("LoginCaptcha")

        loginCaptcha.UserInputID = "CaptchaCode"

        Return loginCaptcha

    End Function

    ' event handler used for Captcha control property randomization
    Public Shared Sub LoginCaptcha_InitializedWebCaptcha( _
        ByVal sender As Object, _
        ByVal e As BotDetect.InitializedWebCaptchaEventArgs)

        If (e.CaptchaId <> "LoginCaptcha") Then
            Return
        End If

        Dim loginCaptcha As Captcha = TryCast(sender, Captcha)

        ' fixed Captcha settings 
        loginCaptcha.ImageSize = New System.Drawing.Size(200, 50)
        loginCaptcha.CodeLength = 4

        ' randomized Captcha settings
        loginCaptcha.ImageStyle = CaptchaRandomization.GetRandomImageStyle()
        loginCaptcha.SoundStyle = CaptchaRandomization.GetRandomSoundStyle()
    End Sub

End Class
